connectivity/FEATURE_BLE/source/generic/GapImpl.cpp

```diff
@@ -1883,15 +1778,15 @@ ble_error_t Gap::setAdvertisingData(
         return BLE_ERROR_INVALID_PARAM;
     }
 
-    if (!_active_sets.get(handle) && payload.size() > this->getMaxActiveSetAdvertisingDataLength()) {
+    if ((_pending_sets.get(handle) || _active_sets.get(handle)) && payload.size() > this->getMaxActiveSetAdvertisingDataLength()) {
```

Comment:
losing the `!` changed the meaning of this if, did you mean `NOT pending AND NOT active`?

Reply:
Yes, the `!` was a bug. 

```
If advertising is currently enabled for the specified advertising set and Operation does not have the value 0x03 or 0x04, the Controller shall return the error code Command Disallowed (0x0C).
```

Values 0x03 and 0x04 are `Complete extended advertising data` and `Unchanged data (just update the Advertising DID)`